tasks/notify.py

```diff
@@ -445,54 +447,128 @@ def update_statistics(job_executions: PipelineRuns):
+        delta = timedelta(days=1) if not allowed_to_fail else timedelta(weeks=1)
+        you_own = ' you own' if channel != '#agent-platform-ops' else ''
+        flaky_tests = (
+            ''
+            if allowed_to_fail
+            else ' In case of tests, you can <https://datadoghq.atlassian.net/wiki/spaces/ADX/pages/3405611398/Flaky+tests+in+go+introducing+flake.Mark|mark them as flaky>.'
+        )
+        expected_to_fail = 'They are allowed to fail' if allowed_to_fail else 'They are not expected to fail'
+
+        message = []
+        for name, fail in stats:
+            link = CI_VISIBILITY_JOB_URL.format(quote(name))
+            message.append(f"- <{link}|{name}>: *{fail} failures*")
+
+        timestamp_start = int((datetime.now() - delta).timestamp() * 1000)
+        timestamp_end = int(datetime.now().timestamp() * 1000)
+
+        header = f'{period} Job Failure Report'
+        description = f'These jobs{you_own} had the most failures in the last {duration}:'
+
+        footer = (
+            f'{expected_to_fail}. Click <https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Ajob%20env%3Aprod%20%40git.repository.id%3A%22gitlab.ddbuild.io%2FDataDog%2Fdatadog-agent%22%20%40ci.pipeline.name%3A%22DataDog%2Fdatadog-agent%22%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amain%20%40ci.status%3Aerror%20%40gitlab.pipeline_source%3A%28push%20OR%20schedule%29%20{not_allowed_query}%40ci.allowed_to_fail%3Atrue&agg_m=count&agg_m_source=base&agg_q=%40ci.job.name&start={timestamp_start}&end={timestamp_end}&agg_q_source=base&agg_t=count&fromUser=false&index=cipipeline&sort_m=count&sort_m_source=base&sort_t=count&top_n=25&top_o=top&viz=toplist&x_missing=true&paused=false|here> for more details.{flaky_tests}\n'
+            + NOTIFICATION_DISCLAIMER
+        )
+
+        body = '\n'.join(message)
+        # Rarely the body may be bigger than 3K characters, split into two messages in this case
+        if len(body) >= 3000:
+            body = ['\n'.join(message[: len(message) // 2]), '\n'.join(message[len(message) // 2 :])]
+        else:
+            body = [body]
+
+        blocks = [
+            {'type': 'header', 'text': {'type': 'plain_text', 'text': header}},
+            {'type': 'section', 'text': {'type': 'mrkdwn', 'text': description}},
+            *[{'type': 'section', 'text': {'type': 'mrkdwn', 'text': text}} for text in body],
+            {'type': 'context', 'elements': [{'type': 'mrkdwn', 'text': ':information_source: ' + footer}]},
+        ]
+
+        # Send message
+        client = WebClient(os.environ["SLACK_API_TOKEN"])
+        client.chat_postMessage(channel=channel, blocks=blocks)
+
+    # Get args passed by the environment variable
     if jobs is None:
-        jobs = os.environ["JOB_FAILURES"]
-        jobs = json.loads(jobs)
+        args = os.environ["ARGS"]
+        args = json.loads(args)
+        jobs = args['jobs']
+        allowed_to_fail = args['allowedToFail']
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Shouldn't we add some logic in case `os.environ["ARGS"]` or `args['jobs']` or `args['allowedToFail']` doesn't exist ?

Reply:
In this case, there will be an error with an explicit stack trace. Do you want to send a message to the agent platform channel ?

Reply:
You're right, I guess it's fine then !